ably/state.go

```diff
@@ -29,7 +29,7 @@ func (st StateType) String() string {
 	case StateChan:
 		return "channel"
 	default:
-		return ""
+		return "invalid"
```

Comment:
> we can't log from the inside of the model. At least I think so.

I understand your point, but that would be a default behaviour across our libraries. We always have a global logger to do such things. Take a look at https://github.com/ably/ably-java/blob/870c33991d0a66b182b7f3196d99c21e39d814a1/src/io/ably/http/TokenAuth.java which uses `Log`, a globally accessible logger (https://github.com/ably/ably-java/blob/245a3f20a6dce0d34413ddfed19c5da8ea647422/src/io/ably/util/Log.java).